### ISSUE TITLE
Attributes name containing dash, e.g. `aria-hidden`

### ISSUE THREAD

**Issue by grepsuzette** [haxetink/tink_domspec#16]
`aria-hidden="true"` is an **attribute** which has the effect of having the marked node and its children ignored by the **accessibilty APIs** (so `<span aria-hidden="true">Not everything should be said</span>` would not be read aloud by text-to-speech systems).

I would like to submit a PR having `aria-hidden` plus a couple of others to address some missing accessibility features, which I think is important both for library support and users.

However I am unsure how,  because of the dash (`aria-hidden`) it contains.

Another related question. This code compiles fine:
```
function render() '
<div class="modal" style="display: block" tabIndex={-1}>
   <div class="modal-dialog">
     <div class="modal-content">
      <div class="modal-header">
         <h5 class="modal-title">{title}</h5>
         <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <!--<span aria-hidden="true">&times;</span>-->
         </button>
           ... etc
';
```
It is puzzling that the commented out `<span aria-hidden..`> would refuse to compile (I think it's because it's missing from the specs). While the `<button .. aria-label="Close">` right above it sees no problem, yet it is absent from the view markup in the DOM inspector.  

Is there an obvious reason why?

**Comment by grepsuzette** [comment]
Hum, regarding 1), found a way... like `@:html('aria-hidden') @:optional var ariaHidden(default, never):Bool;` in `tink.domspec.Attributes.GlobalAttr`, then in the code itself `<span ariaHidden={true}>x</span>`.

Doesn't explain why, with `<button>` in above example, the attributes `data-dismiss` and `aria-label` are accepted even though absent from the specs though. With `<button foo-bar="baz">` it is rejected at compilation, so figure it's maybe defined somewhere else.

Edit: 

As for 2) it seems to be because of the dash. Consider `aria-label="a" data-dismiss="b" data-tabIndex="c" data-bullshit="d" aria-bullshit="e"`. They all will compile except the last one, because its 2 parts around the dash don't contain a recognized attribute name. 
Leaving me now to wonder, is this a feature (this way aria could be defined in one go, as for data) or a bug? In any case I guess it's saner just defining all exact needed attributes even though it's now transparently passed when there is a partial match, right?

**Comment by back2dos** [comment]
It's governed by this RegEx: https://github.com/MVCoconut/coconut.react-dom/blob/a867a88c00a97157a7e2e8af95cef1b7394abd83/src/coconut/react/macros/Html.hx#L45

Data attributes can be arbitrary. Aria attributes are not. But from the looks of it I lost motivation to list them all when I was writing this ;)

**Comment by grepsuzette** [comment]
It looks that only a few of them (Aria) are frequently used. Most seem more or less legacy because the html 5 tags provide more semantics than before. I think I will only propose `aria-hidden, aria-label and aria-labelledby` in a PR for now, if others are useful too they can easily be added later.

**Comment by kLabz** [comment]
I started [something](https://github.com/kLabz/haxe-react/blob/next/src/lib/react/jsx/AriaAttributes.hx) a while ago, don't remember how complete it is. Maybe this could be moved (and improved?) to something we can both use?

**Comment by grepsuzette** [comment]
Hum I think we should wait before including *all* aria attributes.
Because there are about 55 aria attributes, and only about 20 global attributes listed in tink_domspec. 
It would be take tink_domspec too strongly in a certain direction I guess.

OTOH the work you have seems really thorough, so should the need arise it can probably be added very quickly working from it.
